master/buildbot/status/web/build.py

```diff
@@ -361,15 +361,19 @@ def content(self, req, cxt):
             step['link'] = path_to_step(req, s)
             step['text'] = " ".join(s.getText())
             urls = []
-            getUrls = s.getURLs().items()
-            for k,v in s.getURLs().items():
-                if isinstance(v, dict):
-                    if 'results' in v.keys() and v['results'] in css_classes:
-                        url_dict = dict(logname=k, url=v['url'] + codebases_arg, results=css_classes[v['results']])
+
+            # TODO: rather than doing this, it'd be easier to refactor logname
+            for url in s.getURLs():
+                urlName = url["name"]
+                urlLink = url["url"]
+                if "results" in url:
+                    if url["results"] in css_classes:
+                        url_dict = dict(logname=urlName, url=urlLink['url'] + codebases_arg,
+                                        results=css_classes[url['results']])
                     else:
-                        url_dict = dict(logname=k, url=v['url'] + codebases_arg)
+                        url_dict = dict(logname=urlName, url=urlLink['url'] + codebases_arg)
                 else:
-                    url_dict = dict(logname=k, url=v + codebases_arg)
+                    url_dict = dict(logname=urlName, url=urlLink + codebases_arg)
```

Comment:
I propose to be consistent and do not replace 'url' with the 'link', name with the 'logname', etc.I am expecting the following structure